mlflow/pyfunc/__init__.py

```diff
@@ -224,6 +224,8 @@ def predict(*args):
         columns = [str(i) for i, _ in enumerate(args)]
         pdf = pandas.DataFrame(schema, columns=columns)
         result = model.predict(pdf)
+        if type(result) == pandas.DataFrame:
```

Comment:
I think we should use `isinstance(result, pandas.DataFrame)` instead of `type(result) == pandas.DataFrame` here, unless we don't intend to treat subclasses of `pandas.DataFrame` in the same way.

Reply:
Makes sense.